optimum/exporters/neuron/utils.py

```diff
@@ -0,0 +1,156 @@
+        unet_input_shapes (`Dict[str, int]`):
+            Static shapes used for compiling unet.
+        vae_post_quant_conv_input_shapes (`Dict[str, int]`):
+            Static shapes used for compiling vae post quant convolution layer.
+        dynamic_batch_size (`bool`, defaults to `False`):
+            Whether the Neuron compiled model supports dynamic batch size.
+
+    Returns:
+        `Dict[str, Tuple[Union[`PreTrainedModel`, `ModelMixin`], `NeuronConfig`]: A Dict containing the model and
+        Neuron configs for the different components of the model.
+    """
+    models_for_export = {}
+
+    # Text encoder
+    text_encoder = copy.deepcopy(pipeline.text_encoder)
+    text_encoder_config_constructor = TasksManager.get_exporter_config_constructor(
+        model=text_encoder, exporter="neuron", task="feature-extraction"
+    )
+    text_encoder_neuron_config = text_encoder_config_constructor(
+        text_encoder.config,
+        task="feature-extraction",
+        dynamic_batch_size=dynamic_batch_size,
+        **text_encoder_input_shapes,
+    )
+    models_for_export["text_encoder"] = (text_encoder, text_encoder_neuron_config)
+
+    # VAE Decoder
+    vae_config = pipeline.vae.config
+    vae_decoder = copy.deepcopy(pipeline.vae.decoder)
+    vae_decoder_config_constructor = TasksManager.get_exporter_config_constructor(
+        model=vae_decoder,
+        exporter="neuron",
+        task="semantic-segmentation",
+        model_type="vae-decoder",
+    )
+    vae_decoder_neuron_config = vae_decoder_config_constructor(
+        vae_config,
+        task="semantic-segmentation",
+        dynamic_batch_size=dynamic_batch_size,
+        **vae_decoder_input_shapes,
+    )
+    models_for_export["vae_decoder"] = (vae_decoder, vae_decoder_neuron_config)
+
+    # U-NET
+    unet = copy.deepcopy(pipeline.unet)
+    unet_neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+        model=unet, exporter="neuron", task="semantic-segmentation", model_type="unet"
+    )
+    unet_neuron_config = unet_neuron_config_constructor(
+        unet.config,
+        task="semantic-segmentation",
+        dynamic_batch_size=dynamic_batch_size,
+        **unet_input_shapes,
+    )
+    models_for_export["unet"] = (unet, unet_neuron_config)
+
+    # VAE post quant conv
+    post_quant_conv = copy.deepcopy(pipeline.vae.post_quant_conv)
+    post_quant_conv_neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+        model=post_quant_conv, exporter="neuron", task="semantic-segmentation", model_type="conv2d"
+    )
+    post_quant_conv_neuron_config = post_quant_conv_neuron_config_constructor(
+        vae_config,
+        task="semantic-segmentation",
+        dynamic_batch_size=dynamic_batch_size,
+        **vae_post_quant_conv_input_shapes,
+    )
+    models_for_export["vae_conv"] = (post_quant_conv, post_quant_conv_neuron_config)
+
+    del pipeline
```

Comment:
Why?
I think it's bad that a function `del` one of its arguments.

Reply:
It's suggested by aws-neuron-samples to save RAM

```
In order to save RAM (these compiles need lots of RAM!), before tracing each model, we make a deepcopy of the part of the pipeline (i.e. the UNet or the VAE decoder) that is to be traced, and then delete the pipeline object from memory with del pipe. This trick allows the compile to succeed on instance types with a smaller amount of RAM.
```

https://github.com/aws-neuron/aws-neuron-samples/blob/master/torch-neuronx/inference/hf_pretrained_sd2_512_inference.ipynb